test/promise-subclass.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = (tap) => {
+  return class SubclassedPromise extends Promise {
+    then(onSuccess, onReject) {
+      tap.type(onSuccess, 'function');
+      tap.type(onReject, 'undefined');
+      return Promise.prototype.then.call(this, onSuccess, onReject);
```

Comment:
Thanks for the explanation 👍 I'm fine with just merging this then. We can always modify the style in master if we want